### PR TITLE
Fix and update time semantics for PersistentGraph

### PR DESCRIPTION
**Pull request by ljeub-pometry**
### What changes were proposed in this pull request?

- Changed the semantics for edge deletions without a corresponding addition so that they are only considered as an instantaneous event (the edge does not exist before or after)
- Fixed bug where property values for exploded edges were incorrect for the PersistentGraph
- Cleaned up semantics for earliest and latest time on edges accordingly

### Why are the changes needed?

- Bugs and strange/hard to implement behaviour for the PersistentGraph

### Does this PR introduce any user-facing change? If yes is this documented?

Yes

### How was this patch tested?

Added more tests for the semantics and edge cases

### Are there any further changes required?

[ ] Exploded edges and properties have different semantics at the start of the window (needs to be fixed and unified)

[ ] Document the semantics properly

### CODE REVIEW

**Review by miratepuffin — APPROVED** [review]
Couple of random questions

File: raphtory/src/db/graph/views/deletion_graph.rs [hunk 1]
```diff
@@ -66,38 +69,22 @@ fn alive_before<
...
-        (Some(_), None) => false,
-        (None, None) => false,
-    };
-    // None is less than any value (see test below)
-    only_deleted || last_addition_before_start > last_deletion_before_start
+    last_addition_before_start > last_deletion_before_start
 }
 
-fn alive_at<
+fn has_persisted_event<
```
[the remark this answers is not in the document]
Comment (ljeub-pometry) [reply raphtory/src/db/graph/views/deletion_graph.rs:77]: this is not pub

File: raphtory/src/db/graph/views/deletion_graph.rs [hunk 2]
```diff
@@ -491,10 +538,11 @@ impl TimeSemantics for PersistentGraph {
         layer_ids: &LayerIds,
     ) -> Option<i64> {
         let edge = self.core_edge(e.pid());
+        let layer_ids = layer_ids.constrain_from_edge(e);
         match e.time().map(|ti| ti.t()) {
             Some(t) => {
                 let t_start = t.saturating_add(1);
-                edge.updates_par_iter(layer_ids)
```
[the remark this answers is not in the document]
Comment (ljeub-pometry) [reply raphtory/src/db/graph/views/deletion_graph.rs:497]: given past experience, this should be a lot faster

File: raphtory/src/db/graph/views/deletion_graph.rs [hunk 3]
```diff
@@ -1290,4 +1552,281 @@ mod test_deletions {
...
+        let g = PersistentGraph::new();
+        g.delete_edge(10, 0, 1, None).unwrap();
+        assert_eq!(g.edge(0, 1).unwrap().earliest_time(), Some(10));
+        assert_eq!(g.edge(0, 1).unwrap().latest_time(), Some(10));
+    }
+
+    /// Repeated deletions are ignored, only the first one is relevant. Subsequent deletions do not
+    /// create exploded edges
+    #[test]
+    fn test_repeated_deletions() {
```
[the remark this answers is not in the document]
Comment (ljeub-pometry) [reply raphtory/src/db/graph/views/deletion_graph.rs:1611]: Yes it would
